### PR TITLE
Add bindep integration

### PR DESCRIPTION
**Pull request by stephenfin**
### Feature or Bugfix
- Feature

### Purpose
The bindep (BINary DEPendency automation) tool is a tool for checking
the presence of binary packages needed to use an application / library.
Use it to specify the dependencies needed to successfully pass the LaTex
tests. Only Fedora targets are currently provided.

### Relates
This is partial fix for #5823. A more complete fix would be to skip the tests if packages were missing.

### CODE REVIEW

File: bindep.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+texlive-wrapfig [platform:rpm]
+texlive-upquote [platform:rpm]
+texlive-capt-of [platform:rpm]
+texlive-needspace [platform:rpm]
+texlive-polyglossia [platform:rpm]
+texlive-luatex85 [platform:rpm]
+texlive-anyfontsize [platform:rpm]
+texlive-ctablestack [platform:rpm]
+texlive-gnu-freefont [platform:rpm]
+latexmk [platform:rpm]
```
Comment (tk0miya) [inline bindep.txt:15]: Is python needed here?

**Review by tk0miya — APPROVED** [review]
+0; AFAIK no maintainer uses RedHat base OS. So contribution is needed to keep maintained this.

File: bindep.txt (hunk 1 above)
Comment (stephenfin) [reply bindep.txt:15]: It would be unnecessary. `bindep` relies on Python, so without that we can't use bindep :smile:

**Comment by stephenfin** [comment]
> +0; AFAIK no maintainer uses RedHat base OS. So contribution is needed to keep maintained this.

If there is a list of Ubuntu/Debian packages that are required, it's possible to add them to the same file. As far as I can tell, `bindep` is available on Ubuntu.

**Comment by jfbu** [comment]
I am not familiar with Fedora packaging of TeXLive. What happens if there is a local TeXLive installation done directly without the package manager? Will `bindep` then not wrongly complain that some TeX dependency is missing? Personally if I had a Fedora box, I would not rely on the Fedora provided LaTeX packaging but manage my own directly to have the latest one (and keep ones from earlier years).

**Comment by stephenfin** [comment]
> I am not familiar with Fedora packaging of TeXLive. What happens if there is a local TeXLive installation done directly without the package manager? Will `bindep` then not wrongly complain that some TeX dependency is missing? Personally if I had a Fedora box, I would not rely on the Fedora provided LaTeX packaging but manage my own directly to have the latest one (and keep ones from earlier years).

It would, yes. I think if that were the case though, you're already beyond the need for `bindep`?

**Comment by jfbu** [comment]
> It would, yes. I think if that were the case though, you're already beyond the need for `bindep`?

indeed, but only with respect to TeX dependencies... my  remark is rhetorical as I see here that your PR is indeed devoted fully to TeX dependencies.

We very rarely change the TeX dependencies but there will be need for some RedHat based distro-user for maintaining requirement file if that happens...

**Review by jfbu — APPROVED** [review]

File: bindep.txt (hunk 1 above)
Comment (tk0miya) [reply bindep.txt:15]: I see

Comment (tk0miya) [reply bindep.txt:15]: I noticed graphviz and imagemagick are missing.
I think this Dockerfile is helpful for listing up dependencies.
https://github.com/sphinx-doc/docker-ci/blob/master/Dockerfile

**Comment by tk0miya** [comment]
>If there is a list of Ubuntu/Debian packages that are required, it's possible to add them to the same file. As far as I can tell, bindep is available on Ubuntu.

My concern is how to maintain this rpm list. I understand this would be useful for new developers and package maintainers. But I worried it would not be maintained.

File: bindep.txt (hunk 1 above)
Comment (stephenfin) [reply bindep.txt:15]: Good point. I'll add those.

**Comment by tk0miya** [comment]
any comments?

IMO, +0 if anyone will maintain the distro.

**Comment by stephenfin** [comment]
@tk0miya For now, I can maintain the RPM list. Going forward, I'd like to do this via CI and will investigate this. If it turns out to be unmaintained, it's very easy to delete it again.

**Comment by tk0miya** [comment]
Okay, let's merge this!

**Pull request merged**